Hilary's comments on commit id 806d771

Mark: 5/5
You have received full marks for submitting code for comment and using git. 

Good start solving shallow water equations

Some of feedback in the file genericFeedback.txt is relevant to you. Here are some comments that are specific to you. You will need to take this feedback into account for your final code submission in order to achieve a good mark.

0. Good that you are making frequent commits with git

1. You should have a .gitignore file. You should use this to ensure that you do not add large or binary files to your repository such as pdf, pyc or gz files.

2. Good Readme file

3. All python files should have comments at the top describing what is in the file.

4. Good doc-strings

5. You do not need function float_u. Use numpy.zeros instead

6. Your functions for initial conditions should take arrays as functions rather than just single values. See my code in
https://bitbucket.org/mpecdt/mpecdt/
under
source/hilary/code
for an example of how to do this.

7. Your functions for defining whole arrays of initial conditions are very long. Think how you can simplify them and don't include plotting commands inside the functions for initial conditions.

8. nt should always be an integer. You shouldn't need int(nt). If you have declared nt in a way that makes it not an integer then there is a risk that it might take the wrong value when converted to an integer.

9. Use variable names like i,j,k for array indices rather than "x" or "y" as x is usually reserved for distance in the x direction which is a real number rather than an integer.

10. In your code:
for y in range(nx):
    u[y%nx] = uOld[y%nx] - (c*math.sqrt(g/H)/2)*(hOld[(y+1)%nx] - ...

you don't need the %nx when the index is y because you know that y will be in range. It is only y-1 and y+1 that might be out of range.

11. For elements of arrays you do not need to use .copy(). Only for whole arrays

12. I am afraid that I cannot run your code. When I type:
python3 Shallow_water_equations_main.py
I get the error message:
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/matplotlib/__init__.py", line 882, in __setitem__
    cval = self.validate[key](val)
KeyError: 'axes.titlepad'

I think this is to do with the version of matplotlib that I am using. I have tried googling the error message but have not found a solution. What python interpreter and what versions are you using? Include this info in your final code submission and I will have another go at getting your code running on my machine. 

